password-update-service-samples/hibp-integration-ballerina/README.md

```diff
@@ -0,0 +1,147 @@
+# Ballerina-based HIBP Integration for Password Update Service
+
+This is a Ballerina-based password update service that integrates with the HaveIBeenPwned (HIBP) API to check if a password has been compromised in data breaches.
+
+
+> **Note:** This sample is for demonstration only and should not be used in production. It supports the **Pre Update Password Action** of both [Asgardeo](https://console.asgardeo.io/) and [WSO2 Identity Server](https://wso2.com/identity-server/).
+
+## Table of Contents
+- [Overview](#overview)
+- [Features](#features)
+- [Prerequisites](#prerequisites)
+- [Setup Instructions](#setup-instructions)
+  - [1️⃣ Clone the Repository](#1-clone-the-repository)
+  - [2️⃣ Install Dependencies](#2-install-dependencies)
+  - [3️⃣ Run the Service](#3-run-the-service)
+    - [Run the Service Locally](#run-the-service-locally)
+    - [Deploy to Choreo (Required for Asgardeo)](#deploy-to-choreo-required-for-asgardeo)
+  - [4️⃣ Verify the Service](#4-verify-the-service)
+- [Configure the Authenticator in Product](#configure-the-authenticator-in-product)
+  - [In WSO2 Identity Server](#configuring-the-authenticator-in-wso2-identity-server)
+  - [In Asgardeo](#configuring-the-authenticator-in-asgardeo)
+- [API Endpoints](#api-endpoints)
+
+## Overview
+This service implements a custom password update mechanism that checks if a password has been compromised using the HIBP API. It can be used as an external service extension for both Asgardeo and WSO2 Identity Server.
```

Comment:
Throughout let's use password update validation. I will no longer comment on this from this point, so check and update

Reply:
Fixed with 0a6405027d81ee3b63b9bc16b80ededc1973d230